aidb/inference/bound_inference_service.py

```diff
@@ -151,6 +157,7 @@ async def _insert_in_cache_table(self, inp_rows_df: pd.DataFrame, conn):
     # convert the pandas datatype to python native type
     # TODO: can we remove /resolve this?
     inp_rows_df = inp_rows_df.astype('object')
+    inp_rows_df = inp_rows_df.drop_duplicates()
```

Comment:
When sampling with replacement, a single element might be selected multiple times within a batch, potentially leading to insertion errors due to unique key constraints.